src/InitiateState.cpp

```diff
@@ -400,7 +415,7 @@ void* __fastcall do_game_update_new(void* thislol, int edx, int* a, int* b){
 		return do_game_update(thislol, a, b);
 	}
 
-	lua_State* L = (lua_State*)*((void**)thislol);
+	lua_State* L = reinterpret_cast<lua_State *>(thislol);
```

Comment:
It's kind of a hack to get to the first parameter. The signature is correct, it's replicating the 'this' pointer (as in, a pointer to the object), but because I have no idea what the structure is, I can't go this->lua_state etc. I just know that I want whatever is the first thing in the structure. If I wanted the second thing, knowing that the first thing is a pointer, I'd just do *(thislol+sizeof(void*));